crates/router/src/connector/utils.rs

```diff
@@ -105,8 +105,8 @@ impl<Flow, Request, Response> RouterData for types::RouterData<Flow, Request, Re
     where
         T: serde::de::DeserializeOwned,
     {
-        serde_json::from_value::<T>(self.get_connector_meta()?)
-            .into_report()
+        self.get_connector_meta()?
+            .parse_value("SecretSerdeValue")
```

Comment:
The type name would be the type name of `T` here, isn't it?

Something like this:

```suggestion
            .parse_value(std::any::type_name::<T>())
```